src/media/Icon.jsx

```diff
@@ -29,33 +45,32 @@ class Icon extends React.PureComponent {
 			className
 		);
 
-		const viewBox = size === 'auto' ? MEDIA_SIZES['xl'] : MEDIA_SIZES[size];
-		const dim = MEDIA_SIZES[size];
+		const sizeVal = MEDIA_SIZES[size];
 
 		return (
 			<span className={classNames}>
 				<svg
 					preserveAspectRatio='xMinYMin meet'
-					width={dim}
-					height={dim}
-					viewBox={`0 0 ${viewBox} ${viewBox}`}
+					width={sizeVal}
+					height={sizeVal}
+					viewBox={`0 0 ${sizeVal} ${sizeVal}`}
 					className='svg-icon valign--middle'
 					role='img'
 					{...other}>
-					<use xlinkHref={`#icon-${shape}`}></use>
+					<use xlinkHref={`#icon-${getIconShape(shape,size)}`}></use>
```

Comment:
We use an "icon sprite", which means the sprite looks something like this:

`<svg> <symbol id="shape-name" ... <symbol id="another-shape" ...`

We render icons with the `use` tag, which is a pointer to the SVG string contained in the sprite symbol. `xlinkHref` refers to the `id` of the symbol.